controllers/drplacementcontrol.go

```diff
@@ -0,0 +1,1372 @@
+	}
+
+	if homeCluster == "" && d.drpcPlacementRule != nil && len(d.drpcPlacementRule.Status.Decisions) != 0 {
+		homeCluster = d.drpcPlacementRule.Status.Decisions[0].ClusterName
+		homeClusterNamespace = d.drpcPlacementRule.Status.Decisions[0].ClusterNamespace
+	}
+
+	return homeCluster, homeClusterNamespace
+}
+
+// isDeployed check to see if the initial deployment is already complete to this
+// homeCluster or elsewhere
+func (d *DRPCInstance) isDeployed(homeCluster string) (string, bool) {
+	if d.isVRGDeployedAndProtected(homeCluster) {
+		d.log.Info(fmt.Sprintf("Already deployed to %s. Last state: %s",
+			homeCluster, d.getLastDRState()))
+
+		return homeCluster, true
+	}
+
+	clusterName, found := d.isAlreadyDeployedElsewhere(homeCluster)
+	if found {
+		errMsg := fmt.Sprintf("Failed to place deployment on cluster %s, as it is active on another cluster",
+			homeCluster)
+		d.log.Info(errMsg)
+
+		return clusterName, true
+	}
+
+	return "", false
+}
+
+func (d *DRPCInstance) isUserPlRuleUpdated(homeCluster string) bool {
+	return len(d.userPlacementRule.Status.Decisions) > 0 &&
+		d.userPlacementRule.Status.Decisions[0].ClusterName == homeCluster
+}
+
+// isAlreadyFullyDeployedAndProtected will check whether a VRG exists in the targetCluster and
+// whether it is in protected state, and primary.
+func (d *DRPCInstance) isVRGDeployedAndProtected(targetCluster string) bool {
+	d.log.Info(fmt.Sprintf("isAlreadyDeployedAndProtected? - %+v", d.vrgs))
+
+	vrg := d.getCachedVRG(targetCluster)
+	if vrg == nil {
+		return false
+	}
+
+	if !d.isVRGProtected(vrg) {
+		return false
+	}
+
+	return d.isVRGPrimary(vrg)
+}
+
+func (d *DRPCInstance) getCachedVRG(clusterName string) *rmn.VolumeReplicationGroup {
+	vrg, found := d.vrgs[clusterName]
+	if !found {
+		d.log.Info("VRG not found on cluster", "clusterName", clusterName)
+
+		return nil
+	}
+
+	return vrg
+}
+
+func (d *DRPCInstance) isVRGProtected(vrg *rmn.VolumeReplicationGroup) bool {
+	// TODO:
+	// 		UNCOMMENT THIS CODE ONCE PR #255 IS MERGED
+	//
+	// dataProtectedCondition := findCondition(vrg.Status.Conditions, VRGConditionTypeClusterDataProtected)
+	// if dataProtectedCondition == nil {
+	// 	d.log.Info("VRG is not protected yet", "cluster", targetCluster)
+	//
+	// 	return false
+	// }
+	return true
+}
+
+func (d *DRPCInstance) isAlreadyDeployedElsewhere(clusterToSkip string) (string, bool) {
+	for clusterName := range d.vrgs {
+		if clusterName == clusterToSkip {
+			continue
+		}
+
+		return clusterName, true
+	}
+
+	return "", false
+}
+
+func (d *DRPCInstance) startDeploying(homeCluster, homeClusterNamespace string) (bool, error) {
+	const done = true
+
+	// Make sure we record the state that we are deploying
+	d.setDRState(rmn.Deploying)
+	setMetricsTimerFromDRState(rmn.Deploying, timerStart)
+
+	// Create VRG first, to leverage user PlacementRule decision to skip placement and move to cleanup
+	err := d.createVRGManifestWork(homeCluster)
+	if err != nil {
+		return false, err
+	}
+
```

Comment:
(future) wait for VRG to report ClusterDataRestored?

Reply:
That code is for initial deployment.  You are not suggesting to wait for ClusterDataRestore for initial deployment? There should be no ClusterData to restore.

Reply:
Scratch this for now. We will be strict in deploy (as stated in other comments) and so never entertain this state.